src/main.rs

```diff
@@ -0,0 +1,574 @@
+					outputs[0].insert(addr, channel_value_satoshis as f64 / 100_000_000.0);
+					let raw_tx = bitcoind_client.create_raw_transaction(outputs).await;
+
+					// Have your wallet put the inputs into the transaction such that the output is
+					// satisfied.
+					let funded_tx = bitcoind_client.fund_raw_transaction(raw_tx).await;
+					let change_output_position = funded_tx.changepos;
+					assert!(change_output_position == 0 || change_output_position == 1);
+
+					// Sign the final funding transaction and broadcast it.
+					let signed_tx =
+						bitcoind_client.sign_raw_transaction_with_wallet(funded_tx.hex).await;
+					assert_eq!(signed_tx.complete, true);
+					let final_tx: Transaction =
+						encode::deserialize(&hex_utils::to_vec(&signed_tx.hex).unwrap()).unwrap();
+					let outpoint = OutPoint {
+						txid: final_tx.txid(),
+						index: if change_output_position == 0 { 1 } else { 0 },
+					};
+					// Give the funding transaction back to LDK for opening the channel.
+					loop_channel_manager
+						.funding_transaction_generated(&temporary_channel_id, outpoint);
+					pending_txs.insert(outpoint, final_tx);
+				}
+				Event::FundingBroadcastSafe { funding_txo, .. } => {
+					let funding_tx = pending_txs.remove(&funding_txo).unwrap();
+					bitcoind_client.broadcast_transaction(&funding_tx);
+					println!("\nEVENT: broadcasted funding transaction");
+					print!("> ");
+					io::stdout().flush().unwrap();
+				}
+				Event::PaymentReceived { payment_hash, payment_secret, amt: amt_msat } => {
+					let mut payments = payment_storage.lock().unwrap();
+					if let Some((Some(preimage), _, _, _)) = payments.get(&payment_hash) {
+						assert!(loop_channel_manager.claim_funds(
+							preimage.clone(),
+							&payment_secret,
+							amt_msat
+						));
+						println!(
+							"\nEVENT: received payment from payment_hash {} of {} satoshis",
+							hex_utils::hex_str(&payment_hash.0),
+							amt_msat / 1000
+						);
+						print!("> ");
+						io::stdout().flush().unwrap();
+						let (_, _, ref mut status, _) = payments.get_mut(&payment_hash).unwrap();
+						*status = HTLCStatus::Succeeded;
+					} else {
+						println!("\nERROR: we received a payment but didn't know the preimage");
+						print!("> ");
+						io::stdout().flush().unwrap();
+						loop_channel_manager.fail_htlc_backwards(&payment_hash, &payment_secret);
+						payments.insert(
+							payment_hash,
+							(None, HTLCDirection::Inbound, HTLCStatus::Failed, SatoshiAmount(None)),
+						);
+					}
+				}
+				Event::PaymentSent { payment_preimage } => {
+					let hashed = PaymentHash(Sha256::hash(&payment_preimage.0).into_inner());
+					let mut payments = payment_storage.lock().unwrap();
+					for (payment_hash, (preimage_option, _, status, amt_sat)) in payments.iter_mut()
+					{
+						if *payment_hash == hashed {
+							*preimage_option = Some(payment_preimage);
+							*status = HTLCStatus::Succeeded;
```

Comment:
Don't we need to persist the payment info stuff to disk? Is that just a deliberate oversight in the sample (should we comment it) or should we write out the payment info to flat files?

Reply:
Added TODO but yes it should be done at some point